### PR TITLE
Add process language detection enabled in the output of agent status

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR adds process language detection enabled in the output of agent status.

### Motivation
We want to easily see if this feature is enabled. For now we put it under Process Agent.

### Additional Notes
https://github.com/DataDog/datadog-agent/pull/18744 needs to be merged first

### Possible Drawbacks / Trade-offs
None

### Describe how to test/QA your changes
Deploy the agent with and without language detection enabled (set env var`DD_LANGUAGE_DETECTION_ENABLED`). Make sure this setting is correctly shown in the output of `agent status` under `Process Agent` section:
```
❯ k -n datadog-agent-helm exec datadog-agent-linux-dv8lt -- agent status | grep -i language
Defaulted container "agent" out of: agent, trace-agent, process-agent, system-probe, security-agent, init-volume (init), init-config (init), seccomp-setup (init)
2023-08-16 15:50:08 UTC | CORE | WARN | (pkg/util/log/log.go:664 in func1) | Unknown environment variable: DD_PROCESS_CONFIG_LANGUAGE_DETECTION_ENABLED
2023-08-16 15:50:08 UTC | CORE | WARN | (pkg/config/config.go:1578 in LoadCustom) | Unknown environment variable: DD_PROCESS_CONFIG_LANGUAGE_DETECTION_ENABLED
  Process Language Detection Enabled: True
❯ k -n datadog-agent-helm exec datadog-agent-linux-dv8lt -c process-agent -- agent status | grep -i language
2023-08-16 15:52:22 UTC | CORE | WARN | (pkg/util/log/log.go:664 in func1) | Unknown environment variable: DD_PROCESS_CONFIG_LANGUAGE_DETECTION_ENABLED
2023-08-16 15:52:22 UTC | CORE | WARN | (pkg/config/config.go:1578 in LoadCustom) | Unknown environment variable: DD_PROCESS_CONFIG_LANGUAGE_DETECTION_ENABLED
  Process Language Detection Enabled: True
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/status/templates/process-agent.tmpl [hunk 1]
```diff
@@ -25,6 +25,7 @@ Process Agent
   {{- else}}
   System Probe Process Module Status: Not running
   {{- end}}
+  Process Language Detection Enabled: {{ if and .language_detection.enabled (eq .language_detection.enabled "true") }}True{{ else }}False{{ end }}
```
Comment (just-chillin) [inline pkg/status/templates/process-agent.tmpl:28]: Is this exported from the core agent?

Comment (AliDatadog) [reply pkg/status/templates/process-agent.tmpl:28]: I think all the agents have a `status` command

Comment (just-chillin) [reply pkg/status/templates/process-agent.tmpl:28]: Also, if we use sprig we could simplify this to `.language_detection.enabled | title` ([docs](https://masterminds.github.io/sprig/strings.html#:~:text=above%20returns%20hello-,title,-Convert%20to%20title)). It's already in the go mod file, so it should be safe to use

Comment (AliDatadog) [reply pkg/status/templates/process-agent.tmpl]: Apparently it does not work: `panic: template: /process-agent.tmpl:28: function "title" not defined`.

Also I was wrong. This variable is read from a `stats` object which is obtained by sending a request to the process-agent. 

Also, using directly ` {{ expvars.language_detection_enabled }` returns `true` or `false`, without a capital letter.

For testing, I deployed the agent with the environment variable `DD_LANGUAGE_DETECTION_ENABLED=true` in both the agent and process agent. Here are the results:
```
❯ k -n datadog-agent-helm exec datadog-agent-linux-dv8lt -- agent status | grep -i language
Defaulted container "agent" out of: agent, trace-agent, process-agent, system-probe, security-agent, init-volume (init), init-config (init), seccomp-setup (init)
2023-08-16 15:50:08 UTC | CORE | WARN | (pkg/util/log/log.go:664 in func1) | Unknown environment variable: DD_PROCESS_CONFIG_LANGUAGE_DETECTION_ENABLED
2023-08-16 15:50:08 UTC | CORE | WARN | (pkg/config/config.go:1578 in LoadCustom) | Unknown environment variable: DD_PROCESS_CONFIG_LANGUAGE_DETECTION_ENABLED
  Process Language Detection Enabled: True
❯ k -n datadog-agent-helm exec datadog-agent-linux-dv8lt -c process-agent -- agent status | grep -i language
2023-08-16 15:52:22 UTC | CORE | WARN | (pkg/util/log/log.go:664 in func1) | Unknown environment variable: DD_PROCESS_CONFIG_LANGUAGE_DETECTION_ENABLED
2023-08-16 15:52:22 UTC | CORE | WARN | (pkg/config/config.go:1578 in LoadCustom) | Unknown environment variable: DD_PROCESS_CONFIG_LANGUAGE_DETECTION_ENABLED
  Process Language Detection Enabled: True
```

Comment (just-chillin) [reply pkg/status/templates/process-agent.tmpl]: @AliDatadog I think you need to do
```go
template.Funcs(sprig.FuncMap())
)
```

**Review by just-chillin — APPROVED** [review]

File: releasenotes/notes/add-process-language-detection-agent-status-6c0b18fb67c31019.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+enhancements:
+  - |
+    Add ``Process Language Detection Enabled`` in the output of Agent Status under ``Process Agent`` section.
```
Comment (buraizu) [inline releasenotes/notes/add-process-language-detection-agent-status-6c0b18fb67c31019.yaml:3]:
```suggestion
    Add ``Process Language Detection Enabled`` in the output of the Agent Status command under the ``Process Agent`` section.
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with a minor change

**Review by sgnn7 — APPROVED** [review]

File: pkg/status/templates/process-agent.tmpl (hunk 1 above)
Comment (AliDatadog) [reply pkg/status/templates/process-agent.tmpl]: @just-chillin I think the `title` function provided by `sprig` works for string input https://github.com/Masterminds/sprig/blob/e4c0945838d570720d876a6ad9b4568ed32317b4/functions.go#L116

We have a `formatTitle` function but apparently it doesn't work with boolean input so for now I guess it's alright to leave it like this.

**Pull request merged**